src/map/hash_trie_map/mod.rs

```diff
@@ -505,6 +529,31 @@ where
         }
     }
 
+    fn get_mut<Q: ?Sized>(
+        &mut self,
+        key: &Q,
+        key_hash: HashValue,
+    ) -> Option<&mut EntryWithHash<K, V, P>>
+    where
+        K: Borrow<Q>,
+        Q: Hash + Eq,
+    {
+        match self {
+            Bucket::Single(entry) if entry.matches(key, key_hash) => Some(entry),
+            Bucket::Single(_) => None,
+            Bucket::Collision(entries) => {
+                let removed =
+                    bucket_utils::list_remove_first(entries, |e| e.matches(key, key_hash));
```

Comment:
We could do this more efficiently, but I don't think we care, given that collisions are extremely rare.

Reply:
Yeah, it should be unmeasurable in practice. I thought about doing it the "right way" but it would change a lot more code due to the limited api for the list. Doing this way I got away with just "first_mut".